src/Compiler/Service/IncrementalBuild.fs

```diff
@@ -97,88 +94,94 @@ module IncrementalBuilderEventTesting =
+                    |] 
             let diagnosticsLogger = CompilationDiagnosticLogger("Parse", tcConfig.diagnosticsOptions)
             // Return the disposable object that cleans up
             use _holder = new CompilationGlobalsScope(diagnosticsLogger, BuildPhase.Parse)
 
-            try
-                IncrementalBuilderEventTesting.MRU.Add(IncrementalBuilderEventTesting.IBEParsed fileName)
-                let canSkip = sigNameOpt.IsSome && FSharpImplFileSuffixes |> List.exists (FileSystemUtils.checkSuffix fileName)
-                use act =
-                    Activity.start "IncrementalBuildSyntaxTree.parse"
-                        [|
-                            Activity.Tags.fileName, source.FilePath
-                            "buildPhase", BuildPhase.Parse.ToString()
-                            "canSkip", canSkip.ToString()
-                        |]             
-                let input =
-                    if canSkip then
-                        ParsedInput.ImplFile(
-                            ParsedImplFileInput(
-                                fileName,
-                                false,
-                                sigNameOpt.Value,
-                                [],
-                                [],
-                                [],
-                                isLastCompiland,
-                                { ConditionalDirectives = []; CodeComments = [] },
-                                Set.empty
-                            )
-                        )
-                    else
-                        use text = source.GetTextContainer()
-                        match text with
-                        | TextContainer.Stream(stream) ->
-                            ParseOneInputStream(tcConfig, lexResourceManager, fileName, isLastCompiland, diagnosticsLogger, false, stream)
-                        | TextContainer.SourceText(sourceText) ->
-                            ParseOneInputSourceText(tcConfig, lexResourceManager, fileName, isLastCompiland, diagnosticsLogger, sourceText)
-                        | TextContainer.OnDisk ->
-                            ParseOneInputFile(tcConfig, lexResourceManager, fileName, isLastCompiland, diagnosticsLogger, true)
+            try           
+                use text = source.GetTextContainer()
+                let input = 
+                    match text with
+                    | TextContainer.Stream(stream) ->
+                        ParseOneInputStream(tcConfig, lexResourceManager, fileName, isLastCompiland, diagnosticsLogger, false, stream)
+                    | TextContainer.SourceText(sourceText) ->
+                        ParseOneInputSourceText(tcConfig, lexResourceManager, fileName, isLastCompiland, diagnosticsLogger, sourceText)
+                    | TextContainer.OnDisk ->
+                        ParseOneInputFile(tcConfig, lexResourceManager, fileName, isLastCompiland, diagnosticsLogger, true)
```

Comment:
Note that in my tests (in Rider) this `OnDisk` branch was being hit. Does this mean that branch will not be covered by the cache?

Reply:
What is a stamp here?
For file-based requests, can we use file modification timestamps as a hash code, plus `FSharpParsingOptions` as it happens in the existing parsing cache?

Reply:
@auduchinok Would you be able to assess if the approach is/can easily be made compatible with the way Rider queries FCS and would it handle the main usecase where files are type-checked through a call to `ParseAndCheckInputs`?

Reply:
> Note that in my tests (in Rider) this OnDisk branch was being hit. Does this mean that branch will not be covered by the cache?

We should look at how to make the `SourceText` branch work in Rider too. It hasn't been there previously, and we probably need some additional changes now.